extensions/amp-truncate-text/1.0/amp-truncate-text.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CSS as COMPONENT_CSS} from './truncate-text.jss';
+import {CSS} from '../../../build/amp-truncate-text-1.0.css';
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {TruncateText} from './truncate-text';
+import {dict} from '../../../src/utils/object';
+import {isExperimentOn} from '../../../src/experiments';
+import {userAssert} from '../../../src/log';
+
+/** @const {string} */
+const TAG = 'amp-truncate-text';
+
+class AmpTruncateText extends PreactBaseElement {
+  /** @override */
+  init() {
+    this.registerApiAction('expand', (api) => api./*OK*/ expand());
+    this.registerApiAction('collapse', (api) => api./*OK*/ collapse());
+
+    return dict({
+      'onBeforeExpand': this.beforeExpand_.bind(this),
+      'onAfterCollapse': this.afterCollpase_.bind(this),
+    });
+  }
```

Comment:
1. This only runs once, so you can inline the handlers.
2. These would be better as `staticProps`.
3. It's probably simpler to forgo these props and flip a JSS classname (as mentioned above) instead.

Reply:
I kept this attribute approach at the time in order to preserve that part of the API. Is it okay for the Bento version to not use the same sets of classes/attributes? Moving forward with the assumption that it's fine.

Reply:
Class names prefixed with `i-amphtml-` should be fine to exclude from the Preact component since they're supposed to be `i-`nternal: the validator forbids selecting them in authored CSS.

Reply:
Removed in local branch; will push soon